### PR TITLE
Install cancancan

### PR DESCRIPTION
**Pull request by SaboorSohaib**
### Hi Freind :wave: 

**We did the following requirements:**
-  Install CanCanCan in your project.
-  Add a role column to the users' table. Remember to use migration for this.
- A user can delete a post if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.
   - For that, you need to implement the post-deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.
- A user can delete a comment if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan 
    for this authorization.
   - For that, you need to implement the comment-deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.

**Thank You.**

### CODE REVIEW

File: app/models/ability.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class Ability
+  include CanCan::Ability
+  def initialize(user)
+    can :manage, :all if user.role == 'admin'
+
+    can :destroy, Post, author_id: user.id
+    can :destroy, Comment, author_id: user.id
```
Comment (TadesseDev) [inline app/models/ability.rb]: - [ ] Good job adding most of the authorization rules @SaboorSohaib, But  Please remember that owner of the post can perform all view, update or destroy their own post if `Post.author == user` this same rule holds true for comments.

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -26,6 +26,11 @@ def create
...
   end
 
+  def destroy
+    Post.delete(params[:id])
+    redirect_to user_posts_path(params[:user_id])
+  end
+
   private
 
   def post_params
```
Comment (TadesseDev) [inline app/controllers/posts_controller.rb:37]: - [x] For post controller it's among the primary requirement to use **CanCanCan** for authorization. Please add the `load_and_authorize_resource` method of the `CanCanCan` library to your post controller. 
- [Optional] Good job adding the `destroy` method. I suggest once you add the `load_and_authorize_resource`, validate the owner of the post before allowing it to destroy a post.

File: app/controllers/comments_controller.rb [hunk 3]
```diff
@@ -10,4 +10,9 @@ def create
 
     redirect_to request.original_url
   end
+
+  def destroy
+    Comment.delete(params[:id])
+    redirect_to user_post_path(params[:user_id], params[post_id])
+  end
```
Comment (TadesseDev) [inline app/controllers/comments_controller.rb:18]: - [Optional] Good job adding the `destroy` method. I suggest once you add the `load_and_authorize_resource`, validate the owner of the comment before allowing it to destroy a comment.

**Review by TadesseDev — CHANGES_REQUESTED** [review]
### Hi Team

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
212094426-b41dd457-c9db-4ab3-bbe9-176dd8dd08b7


## Highlights
 - [ ]  Correct GitFlow  ✅
 - [ ]  Good commit history  ✅
 - [ ]  Good PR title/description  ✅
 - [ ]  All linters passed  ✅
 
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/ability.rb (hunk 1 above)
Comment (SaboorSohaib) [reply app/models/ability.rb]: ### Hi @TadesseDev 
Thank you very much for reviewing and giving a helpful review.

Comment (TadesseDev) [reply app/models/ability.rb]: I'll try to elaborate more @SaboorSohaib. Well, According to your validation, the owner of a post can only destroy a post. We need to allow users to `manage` posts if he/she is the owner.

Comment (SaboorSohaib) [reply app/models/ability.rb]: **Hi @TadesseDev** 
Ok thank you, I fixed that.

Comment (TadesseDev) [reply app/models/ability.rb]: good job @SaboorSohaib 🙌

**Review by algerina — APPROVED** [review]
Hi @SaboorSohaib  ,

Good job so far! 

May I highlight the following:
- The repo looks professional.:heavy_check_mark:
- Your project looks awesome:heavy_check_mark:

## STATUS: APPROVED :muscle::clap:

Congratulations! :tada:

Your project is complete!  It's time to merge it :shipit:
You did well implementing the previous TSE's requirements



 
 ## Optional suggestions

 - Please check the comment under the review.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread and tag me @algerina if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)
___
_If anything seems to be unclear to you, reach out and ask anything by commenting on this Pull Request and mention me as [Algerina](https://github.com/algerina), I will receive a notification and will respond._

[GitHub](https://github.com/algerina) | [LinkedIn](https://linkedin.com/in/amel-khiri) | [Twitter](https://twitter.com/AmelQahwadji)

_Reviewed by [Amel Khiri ](https://github.com/algerina)_

**Comment by SaboorSohaib** [comment]
> Hi @SaboorSohaib ,
> 
> Good job so far!
> 
> May I highlight the following:
> 
> * The repo looks professional.heavy_check_mark
> * Your project looks awesomeheavy_check_mark
> 
> ## STATUS: APPROVED 
> Congratulations! tada
> 
> Your project is complete! It's time to merge it :shipit: You did well implementing the previous TSE's requirements
> 
>  [ 68747470733a2f2f6d65646961342e67697068792e636f6d2f6d656469612f4c6c596a476939724572624c6a79614d4e722f67697068792e676966 ](https://camo.githubusercontent.com/b70d13b7ae6e012d864c4b3dc969064bfb4975036b3aa32a8fa5d9eabb7dea7e/68747470733a2f2f6d65646961342e67697068792e636f6d2f6d656469612f4c6c596a476939724572624c6a79614d4e722f67697068792e676966) [ ](https://camo.githubusercontent.com/b70d13b7ae6e012d864c4b3dc969064bfb4975036b3aa32a8fa5d9eabb7dea7e/68747470733a2f2f6d65646961342e67697068792e636f6d2f6d656469612f4c6c596a476939724572624c6a79614d4e722f67697068792e676966)
> 
> ## Optional suggestions
> * Please check the comment under the review.
> 
> _Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._ Cheers and Happy coding! clap clap clap
> 
> Feel free to leave any questions or comments in the PR thread and tag me @algerina if something is not 100% clear.
> 
> #### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
> As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)
> 
> _If anything seems to be unclear to you, reach out and ask anything by commenting on this Pull Request and mention me as [Algerina](https://github.com/algerina), I will receive a notification and will respond._
> 
> [GitHub](https://github.com/algerina) | [LinkedIn](https://linkedin.com/in/amel-khiri) | [Twitter](https://twitter.com/AmelQahwadji)
> 
> _Reviewed by [Amel Khiri ](https://github.com/algerina)_

### Hi @algerina 
Thank you for reviewing and approving the project.

**Pull request merged**